### PR TITLE
Added a brief intro do HADDOCK3

### PR DESCRIPTION
**Pull request by joaomcteixeira**
- [x] Your PR is about writing documentation for already existing code :fire:

---

* Added small introduction to haddock3
* generate the docs `tox -e docs` and open with `firefox haddock3-docs/index.html`. The page will appear in the menu on the left.
* text made together with @amjjbonvin 
* links will need to be updated in the future once the documentation is deployed online, either on RTD or bonvinlab.org

### CODE REVIEW

**Review by amjjbonvin — APPROVED** [review]

File: docs/intro.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# A brieft introduction to HADDOCK3
+
+HADDOCK3 is the next generation of integrative modelling software in the
```
Comment (mgiulini) [inline docs/intro.md:3]:
```suggestion
HADDOCK3 is the next generation integrative modelling software in the
```

File: docs/intro.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+long-lasting HADDOCK project. It represents a complete rethinking and rewriting
+of the HADDOCK2.X series, implementing a new way to interact with HADDOCK and
+offering new features to users who can now define custom workflows. 
+
+In the previous HADDOCK2.x versions, users had access to a highly
+parameterisable yet rigid simulation pipeline composed of three mandatory steps:
```
Comment (mgiulini) [inline docs/intro.md:9]: was "water" always mandatory in 2.x?

File: docs/intro.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    * `flexref`: *Semi-flexible refinement using a simulated annealing protocol through molecular dynamics simulations in torsion angle space (`it1` in haddock2.x).*
+    * `emref`: *Refinement by energy minimisation (`itw` EM only in haddock2.4).*
+    * `mdref`: *Refinement by a short molecular dynamics simulation in explicit solvent (`itw` in haddock2.X).*
+* **Scoring modules**
+    * `emscoring`: *scoring of a complex performing a short EM (builds the topology and all missing atoms).*
+    * `mdscoring`: *scoring of a complex performing a short MD in explicit solvent + EM (builds the topology and all missing atoms).*
+* **Analysis modules**
+    * `caprieval`: *Calculates CAPRI metrics (i-RMDS, l-RMSD, Fnat, DockQ) with respect to the top scoring model or reference structure if provided.*
+    * `clusterfcc`: *Clusters models based on the fraction of common contacts (FCC)*
+    * `clusterrmsd`: *Clusters models based on pairwise RMSD matrix calculated with the `rmsdmatrix` module.*
```
Comment (mgiulini) [inline docs/intro.md:59]:
```suggestion
    * `clustfcc`: *Clusters models based on the fraction of common contacts (FCC)*
    * `clustrmsd`: *Clusters models based on pairwise RMSD matrix calculated with the `rmsdmatrix` module.*
```

File: docs/intro.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+To keep HADDOCK3's modules organised, we catalogued them into several
+categories. But, there are no constraints on piping modules of different
+categories.
+
+The main module's categories are "topology", "sampling", "refinement",
+"scoring", and "analysis". There is no limit to how many modules can belong to a
+category. Modules are added as developed, and new categories will be created
+if/when needed. You can access the HADDOCK3 documentation page for the list of
+all categories and modules. Below is a summary of the available modules:
```
Comment (mgiulini) [inline docs/intro.md:41]: as this is not crucial for the user, I would maybe reduce the lenght of this paragraph

**Review by mgiulini — APPROVED** [review]

File: docs/intro.md (hunk 2 above)
Comment (amjjbonvin) [reply docs/intro.md:9]: Not really - may-be instead of water we can rather call it: final refinement

**Review by amjjbonvin — APPROVED** [review]

File: docs/intro.md (hunk 4 above)
Comment (joaomcteixeira) [reply docs/intro.md:41]: They are for the understanding of the project and the different modules. That's how we are presenting them in talks. Let's keep it for now and let the text evolve as we read it more times in the future. Many thanks!

**Pull request merged**